src/wren_compiler.c

```diff
@@ -1381,17 +1381,20 @@ typedef enum
 {
   PREC_NONE,
   PREC_LOWEST,
-  PREC_ASSIGNMENT, // =
-  PREC_LOGIC,      // && ||
-  PREC_EQUALITY,   // == !=
-  PREC_IS,         // is
-  PREC_COMPARISON, // < > <= >=
-  PREC_RANGE,      // .. ...
-  PREC_BITWISE,    // | ^ & << >>
-  PREC_TERM,       // + -
-  PREC_FACTOR,     // * / %
-  PREC_UNARY,      // unary - ! ~
-  PREC_CALL,       // . () []
+  PREC_ASSIGNMENT,    // =
+  PREC_LOGIC,         // && ||
+  PREC_BITWISE_OR,    // |
+  PREC_BITWISE_XOR,   // ^
+  PREC_BITWISE_AND,   // &
```

Comment:
We can and should give these each their own precedence levels, but let's move all of the bitwise operators to between `PREC_COMPARISON` and `PREC_RANGE`.

That's different from C, but I think C's precedence choice there was a mistake. It means you can't do natural things like:

```c
if (flags & SOME_BIT == SOME_BIT) ...
```

Making it higher than range would be useful because the result of a bitwise expression is usually a number, which is a meaningful argument to a range. But given that we don't put spaces around `..`, I think it would read confusingly if bitwise operators had higher precedence:

```c
a & b..c
```

I think most people would parse that as `a & (b..c)`, not `(a & b)..c`.